samples/sensor/stream_fifo/src/main.c

```diff
@@ -0,0 +1,247 @@
+			c = decoder->decode(buf, accel_chan, &accel_fit, 8, accel_data);
+
+			for (int k = 0; k < c; k++) {
+				printk("XL data for %s %lluns (%" PRIq(6) ", %" PRIq(6)
+				       ", %" PRIq(6) ")\n", dev->name,
+				       PRIsensor_three_axis_data_arg(*accel_data, k));
+			}
+			i += c;
+
+			/* decode and print Gyroscope FIFO frames */
+			c = decoder->decode(buf, gyro_chan, &gyro_fit, 8, gyro_data);
+
+			for (int k = 0; k < c; k++) {
+				printk("GY data for %s %lluns (%" PRIq(6) ", %" PRIq(6)
+				       ", %" PRIq(6) ")\n", dev->name,
+				       PRIsensor_three_axis_data_arg(*gyro_data, k));
+			}
+			i += c;
+
+			/* decode and print Temperature FIFO frames */
+			c = decoder->decode(buf, temp_chan, &temp_fit, 4, temp_data);
+
+			for (int k = 0; k < c; k++) {
+				printk("TP data for %s %lluns %s%d.%d °C\n", dev->name,
+				       PRIsensor_q31_data_arg(*temp_data, k));
+			}
+			i += c;
+
+			/* decode and print Game Rotation Vector FIFO frames */
+			c = decoder->decode(buf, rot_vector_chan, &rot_vect_fit, 8, rot_vect_data);
+
+			for (int k = 0; k < c; k++) {
+				printk("ROT data for %s %lluns (%" PRIq(6) ", %" PRIq(6)
+				       ", %" PRIq(6) ", %" PRIq(6) ")\n", dev->name,
+				       PRIsensor_game_rotation_vector_data_arg(*rot_vect_data, k));
+			}
+			i += c;
+
+			/* decode and print Gravity Vector FIFO frames */
+			c = decoder->decode(buf, gravity_chan, &gravity_fit, 8, gravity_data);
+
+			for (int k = 0; k < c; k++) {
+				printk("GV data for %s %lluns (%" PRIq(6) ", %" PRIq(6)
+				       ", %" PRIq(6) ")\n", dev->name,
+				       PRIsensor_three_axis_data_arg(*gravity_data, k));
+			}
+			i += c;
+
+			/* decode and print Gyroscope GBIAS FIFO frames */
+			c = decoder->decode(buf, gbias_chan, &gbias_fit, 8, gbias_data);
+
+			for (int k = 0; k < c; k++) {
+				printk("GY GBIAS data for %s %lluns (%" PRIq(6) ", %" PRIq(6)
+				       ", %" PRIq(6) ")\n", dev->name,
+				       PRIsensor_three_axis_data_arg(*gbias_data, k));
+			}
+			i += c;
+		}
+
+		rtio_release_buffer(&stream_ctx, buf, buf_len);
+	}
+}
+
+static void check_sensor_is_off(const struct device *dev)
+{
+	int ret;
+	struct sensor_value odr;
+
+	ret = sensor_attr_get(dev, SENSOR_CHAN_ACCEL_XYZ, SENSOR_ATTR_SAMPLING_FREQUENCY, &odr);
+
+	/* Check if accel is off */
+	if (ret != 0 || (odr.val1 == 0 && odr.val2 == 0)) {
+		printk("%s WRN : accelerometer device is off\n", dev->name);
+	}
+
+	ret = sensor_attr_get(dev, SENSOR_CHAN_GYRO_XYZ, SENSOR_ATTR_SAMPLING_FREQUENCY, &odr);
+
+	/* Check if gyro is off */
+	if (ret != 0 || (odr.val1 == 0 && odr.val2 == 0)) {
+		printk("%s WRN : gyroscope device is off\n", dev->name);
+	}
+}
+
+int main(void)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(sensors); i++) {
+		if (!device_is_ready(sensors[i])) {
+			printk("sensor: device %s not ready.\n", sensors[i]->name);
+			return 0;
+		}
+		check_sensor_is_off(sensors[i]);
+
+		k_thread_create(&thread_id[i], thread_stack[i], TASK_STACK_SIZE, print_stream,
```

Comment:
A thread per sensor shouldn't be needed really, but as a sample its fine

Reply:
Correct, it is just a sample to demonstrate the usage